src/zarr/core/array.py

```diff
@@ -2056,3 +2048,6 @@ def info(self) -> None:
         return sync(
             self._async_array.info(),
         )
+
+
+__all__ = ["Array", "AsyncArray"]
```

Comment:
These are also defined in the `zarr` namespace - they should either be in `zarr` or `zarr.core.array`, not both, so it is clear where users are expected to import them from. I think this comment applies to other __all__ sections too.

Reply:
In [4725370](https://github.com/zarr-developers/zarr-python/pull/2072/commits/4725370095554edde9948c5e5b88ca5452c64977), I've removed most duplicative `__all__` declarations. The notable execeptions are in `creation.py` and `convenience.py` which are both deprecated.